src/mlpack/methods/reinforcement_learning/replay/episodic_replay.hpp

```diff
@@ -0,0 +1,271 @@
+#ifndef MLPACK_METHODS_RL_REPLAY_EPISODIC_REPLAY_HPP
+#define MLPACK_METHODS_RL_REPLAY_EPISODIC_REPLAY_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/math/random.hpp>
+
+namespace mlpack {
+namespace rl {
+
+/**
+ * Implementation of episodic experience replay.
+ *
+ * Each episode of interactions between the agent and the
+ * environment will be saved to a memory buffer. When necessary,
+ * we can simply sample previous episodes or the most recent episode
+ * from the buffer to train the agent.
+ * @tparam EnvironmentType Desired task.
+ */
+template <typename EnvironmentType>
+class EpisodicReplay
+{
+ public:
+  //! Convenient typedef for action.
+  using ActionType = typename EnvironmentType::Action;
+
+  //! Convenient typedef for state.
+  using StateType = typename EnvironmentType::State;
+
+  /**
+  * Construct an instance of episode experience replay class.
+  * Everything initialized to 0
+  */
+  EpisodicReplay():
+      capacity(0),
+      position(0),
+      max_episode_len(0),
+      full(false)
+  { /* Nothing to do here. */ }
+
+  /**
+  * Construct an instance of episode experience replay class.
+  * @param capacity Total memory size in terms of number of episodes.
+  * @param dimension The dimension of an encoded state.
+  */
+  EpisodicReplay(const size_t capacity,
+                 const size_t max_episode_len,
```

Comment:
Looks like a parameter description is missing, also please use camel casing for all names, see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#naming-conventions for more details.